page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/PageManager.java

```diff
@@ -0,0 +1,246 @@
+     * @throws PageInitializationException if failed to execute corresponding page constructor
+     */
+    public static Page getPage(String packageName, String title, WebDriver driver) throws PageInitializationException {
+        return bootstrapPage(getPageClass(packageName, title), driver);
+    }
+
+    /**
+     * Initialize page with specified title and save its instance to
+     * {@link PageContext#currentPage} for further use
+     *
+     * @param title a page title
+     * @return the page instance
+     * @throws PageInitializationException if failed to execute corresponding page constructor
+     */
+    public static Page getPage(String title) throws PageInitializationException {
+        if (null == PageContext.getCurrentPage() || !PageContext.getCurrentPageTitle().equals(title)) {
+            if (null != PageContext.getCurrentPage()) {
+                getPage(PROPERTIES.getPagesPackage(), title, PageContext.getCurrentPage().getDriver());
+            }
+            if (null == PageContext.getCurrentPage()) {
+                getPage(PROPERTIES.getPagesPackage(), title, Environment.getDriverService().getDriver());
+            }
+            if (null == PageContext.getCurrentPage()) {
+                throw new AutotestError("Page object with title '" + title + "' is not registered");
+            }
+        }
+        return PageContext.getCurrentPage();
+    }
+
+    /**
+     * Run constructor of specified page class and put its instance into static
+     * {@link PageContext#currentPage} variable
+     *
+     * @param page a page class
+     * @return the initialized page object
+     * @throws PageInitializationException if failed to execute corresponding page constructor
+     */
+    private static Page bootstrapPage(Class<?> page, WebDriver driver) throws PageInitializationException {
+        if (page != null) {
+            try {
+                @SuppressWarnings("unchecked")
+                Constructor<Page> constructor = ((Constructor<Page>) page.getConstructor(WebDriver.class));
+                constructor.setAccessible(true);
+                Page currentPage = constructor.newInstance(driver);
+                PageContext.setCurrentPage(currentPage);
+                return currentPage;
+            } catch (NoSuchMethodException | SecurityException | InstantiationException | IllegalAccessException | IllegalArgumentException | InvocationTargetException e) {
+                throw new PageInitializationException("Failed to initialize page '" + page + "'", e);
+            }
+        }
+        return null;
+    }
+
+    /**
+     * @param packageName a path to page objects
+     * @param title a page title
+     * @return the page class
+     */
+    private static Class<?> getPageClass(final String packageName, String title) {
+        ClassLoader loader = Thread.currentThread().getContextClassLoader();
+        final Set<Class<?>> allClasses = new HashSet<>();
+        // TODO: var add cash 5/29/17
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)